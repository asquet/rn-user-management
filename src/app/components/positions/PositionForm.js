import React from 'react';
import { View } from 'react-native';
import InputRow from '../common/form/InputRow';
import Button from '../common/content/Button';
import MultiSelect from '../common/form/MultiSelect';
import { Text, Heading } from '../common/content/Text';
import { formStyles } from '../../stylesheet';

export default function PositionForm({ isNew, position, onChange, onSave, onCancel, roles }) {
  return (
    <View style={formStyles.container}>
      <Heading>
        { isNew ? 'Create position' : 'Edit position'}
      </Heading>
      <InputRow
        value={position.name}
        onChange={val => onChange('name', val)}
        label="Name"
      />
      <InputRow label="roles">
        <MultiSelect
          options={roles.map(role => ({ key: role.id, name: role.name }))}
          renderRow={(row, isSelected) => <Text>{row.name} {isSelected ? ' + ' : ' - '}</Text>}
          onSelectionChange={
            (selectedRow, allSelectedRows) => onChange('roles', allSelectedRows)
          }
          selectedOptions={position.roles}
          rowStyle={{ backgroundColor: 'skyblue' }}
        />
      </InputRow>
      <View style={formStyles.controlsWrapper}>
        <Button
          onPress={() => onSave(position)}
          title="Save"
          buttonStyle={formStyles.controlsSave}
        />
        <Button
          onPress={onCancel}
          title="Cancel"
          buttonStyle={formStyles.controlsCancel}
        />
      </View>
    </View>
  );
}


PositionForm.propTypes = {
  isNew: React.PropTypes.bool.isRequired,
  position: React.PropTypes.shape({
    name: React.PropTypes.string,
    roles: React.PropTypes.arrayOf(React.PropTypes.number),
  }).isRequired,
  onChange: React.PropTypes.func.isRequired,
  onSave: React.PropTypes.func.isRequired,
  onCancel: React.PropTypes.func.isRequired,
  roles: React.PropTypes.arrayOf(React.PropTypes.shape({
    id: React.PropTypes.number,
    name: React.PropTypes.string,
  })).isRequired,
};
