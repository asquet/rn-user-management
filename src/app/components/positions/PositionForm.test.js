import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import PositionForm from './PositionForm';

// Note: test renderer must be required after react-native.

describe('PositionForm', () => {
  const position = { name: 'pos' };
  const roles = [{ id: 1, name: '1' }];

  it('renders correctly', () => {
    const tree = renderer.create(
      <PositionForm
        isNew
        onCancel={jest.fn()}
        onChange={jest.fn()}
        onSave={jest.fn()}
        position={position}
        roles={roles}
      />,
    );
    expect(tree.toJSON()).toMatchSnapshot();
  });
});
