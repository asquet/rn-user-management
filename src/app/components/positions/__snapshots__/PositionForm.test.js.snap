// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PositionForm renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "white",
      "flex": 1,
      "flexDirection": "column",
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "color": "#707070",
          "fontSize": 24,
        },
        null,
      ]
    }
  >
    Create position
  </Text>
  <View
    style={
      Array [
        Object {
          "height": 64,
          "marginBottom": 10,
          "padding": 3,
        },
        null,
      ]
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "#707070",
            "fontSize": 18,
          },
          Object {
            "color": "#707070",
            "fontSize": 16,
            "height": 24,
            "marginBottom": 0,
          },
        ]
      }
    >
      Name
    </Text>
    <TextInput
      onChangeText={[Function]}
      placeholder="Name"
      style={
        Object {
          "borderBottomWidth": 1,
          "borderColor": "#C0C0C0",
          "fontSize": 18,
          "height": 40,
        }
      }
      value="pos"
    />
  </View>
  <View
    style={
      Array [
        Object {
          "height": 64,
          "marginBottom": 10,
          "padding": 3,
        },
        null,
      ]
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "#707070",
            "fontSize": 18,
          },
          Object {
            "color": "#707070",
            "fontSize": 16,
            "height": 24,
            "marginBottom": 0,
          },
        ]
      }
    >
      roles
    </Text>
    <RCTScrollView
      dataSource={
        ListViewDataSource {
          "items": 1,
        }
      }
      enableEmptySections={true}
      renderRow={[Function]}
      renderScrollComponent={[Function]}
    >
      <View>
        <View
          accessibilityComponentType={undefined}
          accessibilityLabel={undefined}
          accessibilityTraits={undefined}
          accessible={true}
          hasTVPreferredFocus={undefined}
          hitSlop={undefined}
          isTVSelectable={true}
          onLayout={undefined}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Array [
              Object {
                "backgroundColor": "transparent",
              },
              undefined,
            ]
          }
          testID={undefined}
          tvParallaxProperties={undefined}
        >
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "flexDirection": "row",
                },
                Object {
                  "backgroundColor": "skyblue",
                },
              ]
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Array [
                  Object {
                    "color": "#707070",
                    "fontSize": 18,
                  },
                  null,
                ]
              }
            >
              1
               
               - 
            </Text>
          </View>
        </View>
      </View>
    </RCTScrollView>
  </View>
  <View
    style={
      Object {
        "flexDirection": "row",
      }
    }
  >
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hasTVPreferredFocus={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
          },
          undefined,
        ]
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "backgroundColor": "#9E9E9E",
              "flexDirection": "row",
              "justifyContent": "center",
              "marginLeft": 15,
              "marginRight": 15,
              "padding": 19,
            },
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            Object {
              "padding": 12,
            },
            Array [
              Object {
                "alignItems": "center",
                "borderRadius": 3,
                "padding": 5,
              },
              Object {
                "backgroundColor": "#33CC55",
                "flex": 1,
              },
            ],
            undefined,
            undefined,
          ]
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "fontFamily": undefined,
              },
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              Array [
                Object {
                  "color": "white",
                  "fontSize": 20,
                },
                undefined,
                Object {
                  "fontSize": 17.5,
                },
                undefined,
                undefined,
                undefined,
                undefined,
              ],
            ]
          }
        >
          Save
        </Text>
      </View>
    </View>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hasTVPreferredFocus={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
          },
          undefined,
        ]
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "backgroundColor": "#9E9E9E",
              "flexDirection": "row",
              "justifyContent": "center",
              "marginLeft": 15,
              "marginRight": 15,
              "padding": 19,
            },
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            Object {
              "padding": 12,
            },
            Array [
              Object {
                "alignItems": "center",
                "borderRadius": 3,
                "padding": 5,
              },
              Object {
                "backgroundColor": "#dd5555",
                "flex": 1,
              },
            ],
            undefined,
            undefined,
          ]
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "fontFamily": undefined,
              },
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              Array [
                Object {
                  "color": "white",
                  "fontSize": 20,
                },
                undefined,
                Object {
                  "fontSize": 17.5,
                },
                undefined,
                undefined,
                undefined,
                undefined,
              ],
            ]
          }
        >
          Cancel
        </Text>
      </View>
    </View>
  </View>
</View>
`;
